joblib/test/test_parallel.py

```diff
@@ -1295,8 +1295,10 @@ def test_parallel_unordered_generator_returns_fastest_first(backend, n_jobs):
 def test_parallel_unordered_generator_returns_fastest_first_with_dask(
         n_jobs, context
 ):
-    client = distributed.Client(n_workers=2, threads_per_worker=2)  # noqa
-    with context("dask"):
+    with (
+        distributed.Client(n_workers=2, threads_per_worker=2),
+        context("dask")
+    ):
```

Comment:
It seems that tuple-with expressions do not work with Python 3.8 which we still support for now:

```python-traceback
_ test_deadlock_with_generator_and_dask[parallel_backend-generator_unordered-2] _

context = <class 'joblib.parallel.parallel_backend'>
return_as = 'generator_unordered', n_jobs = 2

    @with_numpy
    @pytest.mark.parametrize('n_jobs', [2, -1])
    @parametrize('return_as', ["generator", "generator_unordered"])
    @parametrize("context", [parallel_config, parallel_backend])
    @skipif(distributed is None, reason='This test requires dask')
    def test_deadlock_with_generator_and_dask(context, return_as, n_jobs):
>       with (
            distributed.Client(n_workers=2, threads_per_worker=2),
            context("dask")
        ):
E       AttributeError: __enter__
```

I think we need nested with expressions for now.